internal/server/middleware/span.go

```diff
@@ -0,0 +1,156 @@
+package middleware
+
+// Taken from: middleware/tracing/span.go
+
+import (
+	"context"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/go-kratos/kratos/v2/metadata"
+	"github.com/go-kratos/kratos/v2/transport"
+	"github.com/go-kratos/kratos/v2/transport/http"
+
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/protobuf/proto"
+)
+
+const serviceHeader = "x-md-service-name" //Taken from Kratos: middleware/tracing/metadata.go
+
+func setClientSpan(ctx context.Context, span trace.Span, m interface{}) {
```

Comment:
Looks like the linter is complaining because setClientSpan() is never called and thus parseTarget() is also never called